frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -3227,6 +3207,64 @@ const IR::Node* TypeInference::postorder(IR::KeyElement* elem) {
     return elem;
 }
 
+void TypeInference::validateActionInitializer(
+    const IR::Expression* actionCall,
+    const IR::P4Table* table) {
+
+    auto al = table->getActionList();
+    if (al == nullptr) {
+        typeError("%1% has no action list, so it cannot have %2%",
+                  table, actionCall);
+        return;
+    }
+
+    auto call = actionCall->to<IR::MethodCallExpression>();
+    CHECK_NULL(call);
```

Comment:
If the expression is not a MethodCallExpression, should that be a error rather than a BUG_CHECK?

Reply:
The pass createBuiltins will convert other expressions to method calls in this context.

Reply:
But maybe there are other things that could happen, so an error would not be bad either.